client/src/ExtensionState.ts

```diff
@@ -112,13 +111,18 @@ export class State {
 
     this.client = new LanguageClient('gobraServer', 'Gobra Server', serverOptions, clientOptions);
 
+    if (this.client == null) {
+      return Promise.reject("languge client couldn't be created");
+    }
+
     // Start the client together with the server.
-    this.disposableServer = this.client.start();
+    const disposable = this.client.start();
+    // Push the disposable to the context's subscriptions so that the
+	  // client can be deactivated on extension deactivation
+    context.subscriptions.push(disposable);
+    this.context = context;
 
-    // check whether the server has started
-    if (!this.disposableServer) {
-      console.log("Error: Failed to start the client.");
-    }
+    await State.client.onReady();
```

Comment:
This is the last operation, you can just do `return State.client.onReady();`, returning the `Promise` back to the caller. Then you can also make the function not be `async`.